plugin/src/main/java/org/opensearch/ml/sdkclient/DDBOpenSearchClient.java

```diff
@@ -352,6 +371,138 @@ public CompletionStage<DeleteDataObjectResponse> deleteDataObjectAsync(
         }), executor);
     }
 
+    @Override
+    public CompletionStage<BulkDataObjectResponse> bulkDataObjectAsync(
+        BulkDataObjectRequest request,
+        Executor executor,
+        Boolean isMultiTenancyEnabled
+    ) {
+        return CompletableFuture.supplyAsync(() -> AccessController.doPrivileged((PrivilegedAction<BulkDataObjectResponse>) () -> {
+            log.info("Performing {} bulk actions on table {}", request.requests().size(), request.getIndices());
+
+            List<DataObjectResponse> responses = new ArrayList<>();
+
+            long startNanos = System.nanoTime();
+            for (DataObjectRequest dataObjectRequest : request.requests()) {
+                try {
+                    if (dataObjectRequest instanceof PutDataObjectRequest) {
+                        responses
+                            .add(
+                                putDataObjectAsync((PutDataObjectRequest) dataObjectRequest, executor, isMultiTenancyEnabled)
```

Comment:
> Dynamo DB supports bulk operations

Only for put and delete.  I had actually started trying to implement it that way with conditionals on whether we were updating, but it got rather complex very quickly with the proper handling of `seqNo`.  Since we are only using it for update we wouldn't even use that code.

I agree that in the long term we should use that API when there are no update requests.

Reply:
(I also noted this in item 5 of the PR description.  Added a comment in the latest commit.)